### ISSUE TITLE
Specifying .ci.yaml iOS simulator runtimes does not reliably install the correct version

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#100921]
I ran the tests for the same PR three times:

1st and 3rd time passed: 
https://ci.chromium.org/p/flutter/builders/try/Mac%20Web%20Engine/23967
https://ci.chromium.org/old/p/flutter/builders/try/Mac%20Web%20Engine/24005

Second time the `Run tests on iOS Safari` tasks timed out (and I eventually cancelled the build):
https://ci.chromium.org/p/flutter/builders/try/Mac%20Web%20Engine/23994

The passing times show that iOS 15.0 is correctly installed, but the failing one has 15.2.

`install xcode runtime ios-15-0 (2)`
```
Executing command [
  '/opt/s/w/ir/cache/osx_sdk/mac_toolchain',
  'install-runtime',
  '-cipd-package-prefix',
  'flutter_internal/ios/xcode',
  '-runtime-version',
  'ios-15-0',
  '-output-dir',
  '/opt/s/w/ir/cache/xcode_runtime_ios-15-0',
]
```
stdout
```
Packages:
  flutter_internal/ios/xcode/ios_runtime:bSZkcV-gse3F7uV37terlLhxOfepuLZVlOOQx6082KEC
```
`( 350 ms )Moving ios-15-0 to final dest (2)`
```
Executing command [
  '/opt/s/w/ir/cipd_bin_packages/vpython',
  '-u',
  '/opt/s/w/ir/kitchen-checkout/recipe_engine/recipe_modules/file/resources/fileutil.py',
  '--json-output',
  '/opt/s/w/ir/x/t/tmpdgirowom.json',
  'symlink',
  '/opt/s/w/ir/cache/xcode_runtime_ios-15-0/iOS 15.0.simruntime',
  '/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 15.0.simruntime',
]
```
`list simulators (2)` shows iOS 15.0 runtime is installed:
```
== Runtimes ==
iOS 13.0 (13.0 - 17A577) - com.apple.CoreSimulator.SimRuntime.iOS-13-0
iOS 15.0 (15.0 - 19A339) - com.apple.CoreSimulator.SimRuntime.iOS-15-0
```

However the failing one simulator does not have 15.0, but 15.2 (the default for that version of Xcode):
```
== Runtimes ==
iOS 13.0 (13.0 - 17A577) - com.apple.CoreSimulator.SimRuntime.iOS-13-0
iOS 15.2 (15.2 - 19C51) - com.apple.CoreSimulator.SimRuntime.iOS-15-2
== Devices ==
```

Blocking https://github.com/flutter/engine/pull/31915
/cc @godofredoc

**Comment by godofredoc** [comment]
There are few complications in  how xcode manage dependencies. The current recipe implementation does not delete all the simruntimes because it was assumed that they could coexist. Turns out that if we have 15.0 and 15.2 runtimes available 15.2 will always take precedence.

There is another catch, we can not delete the default runtime and if it is 15.2 then installing 15.0 won't have any effect for that version of xcode e.g. 13c100 won't work for runtime 15.0.

**Comment by godofredoc** [comment]
Fixed with https://flutter-review.googlesource.com/c/recipes/+/28684
